os/os.module.go

```diff
@@ -0,0 +1,241 @@
+package os
+
+// Copyright 2022 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+import (
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+
+	"github.com/go-python/gpython/py"
+)
+
+func init() {
+
+	methods := []*py.Method{
+		py.MustNewMethod("getcwd", getCwd, 0, "Get the current working directory"),
+		py.MustNewMethod("getcwdb", getCwdb, 0, "Get the current working directory in a byte slice"),
+		py.MustNewMethod("chdir", chdir, 0, "Change the current working directory"),
+		py.MustNewMethod("getenv", getenv, 0, "Return the value of the environment variable key if it exists, or default if it doesn’t. key, default and the result are str."),
+		py.MustNewMethod("getpid", getpid, 0, "Return the current process id."),
+		py.MustNewMethod("putenv", putenv, 0, "Set the environment variable named key to the string value."),
+		py.MustNewMethod("unsetenv", unsetenv, 0, "Unset (delete) the environment variable named key."),
+		py.MustNewMethod("_exit", _exit, 0, "Immediate program termination."),
+		py.MustNewMethod("system", system, 0, "Run shell commands, prints stdout directly to deault"),
+	}
+
+	globals := py.StringDict{
+		"error":   py.OSError,
+		"environ": getEnvVariables(),
+	}
+
+	py.RegisterModule(&py.ModuleImpl{
+		Info: py.ModuleInfo{
+			Name: "os",
+			Doc:  "Miscellaneous operating system interfaces",
+		},
+		Methods: methods,
+		Globals: globals,
+	})
+}
+
+// gets and parses all process enviroment variables
+func getEnvVariables() py.StringDict {
+	env_variables := os.Environ() // this gets all the enviroment variables
+	dict := py.NewStringDict()
+	for _, evar := range env_variables {
+		key_value := strings.SplitN(evar, "=", 2) // returns a []string containing [key,value]
+		dict.M__setitem__(py.String(key_value[0]), py.String(key_value[1]))
+	}
+
+	return dict
+}
+
+// gets the current working directory
+func getCwd(self py.Object, args py.Tuple) (py.Object, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nil, py.ExceptionNewf(py.OSError, "Unable to get current working directory.")
+	}
+	return py.String(dir), nil
+}
+
+// gets the current working directory in a byte list
+func getCwdb(self py.Object, args py.Tuple) (py.Object, error) {
+	dir, err := os.Getwd()
+	if err != nil {
+		return nil, py.ExceptionNewf(py.OSError, "Unable to get current working directory.")
+	}
+	return py.Bytes([]byte(dir)), nil
+}
+
+// change current working directory
+func chdir(self py.Object, args py.Tuple) (py.Object, error) {
+	if len(args) == 0 {
+		return nil, py.ExceptionNewf(py.TypeError, "Missing required argument 'path' (pos 1)")
+	}
+	if objectIsString(args[0]) {
```

Comment:
following Go good practices, we should inverse the logic:
```suggestion
	if !objectIsString(args[0]) {
	    return nil, py.ExceptionNewf(py.TypeError, "str expected, not "+args[0].Type().Name)
	}
```